core/src/main/java/org/apache/brooklyn/core/objs/AbstractEntityAdjunct.java

```diff
@@ -421,6 +421,7 @@ public void setDisplayName(String name) {
     public void setEntity(EntityLocal entity) {
         if (destroyed.get()) throw new IllegalStateException("Cannot set entity on a destroyed entity adjunct");
         this.entity = entity;
+        this.execution = ((EntityInternal) entity).getExecutionContext();
```

Comment:
I'm guessing you hit a problem that made you add this line - can we capture in a test what that problem was, rather than just fixing it?

I'm fine with merging this time, but for future reference I am really keen that we improve our test coverage for when we hit problems like this.